soci/soci_index.go

```diff
@@ -330,32 +387,27 @@ func WriteSociIndex(ctx context.Context, indexWithMetadata IndexWithMetadata, st
 
 	log.G(ctx).WithField("digest", dgst.String()).Debugf("soci index has been written")
 
+	var refers *ocispec.Descriptor
+	if indexWithMetadata.Index.MediaType == OCIArtifactManifestMediaType {
```

Comment:
maybe add a helper function to `Index` to get refers so that we can try to contain the logic in one place behind a function.  Something like : 

```
func (index Index) getRefers() ocispec.Descriptor {
  switch index.MediaType:
    case OCIArtifactManifestMediaType :
.....
}
```

Reply:
Done.